src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,38 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add movie " + movie + " to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
```

Comment:
You don't need to set an ID for your entity explicitly as it will be done by Hibernate. Ensure that the Movie entity is properly annotated with @Id and the generation strategy is set.